src/com/sap/piper/cm/ChangeManagement.groovy

```diff
@@ -172,29 +172,150 @@ public class ChangeManagement implements Serializable {
+
+        if (!deployConfigFile) {
+            useConfigFile = false
+            noConfig = !script.fileExists('ui5-deploy.yaml')
+        } else {
+            if (script.fileExists(deployConfigFile)) {
+              // in this case we will use the config file
+              useConfigFile = true
+            } else {
+                if (deployConfigFile == 'ui5-deploy.yaml') {
+                    // in this case this is most likely provided by the piper default config and
+                    // it was not explicitly configured. Hence we assume not having a config file
+                    useConfigFile = false
+                    noConfig = true
+                } else {
+                    script.error("Configured deploy config file '${deployConfigFile}' does not exists.")
+                }
+            }
+        }
+
+        if (noConfig) {
+            params += ['--noConfig'] // no config file, but we will provide our parameters
+        }
+
+        if (useConfigFile) {
+             params += ['-c', "\"" + deployConfigFile + "\""]
+        }
+
+        //
+        // All the parameters below encapsulated in an if statement might also be provided in a config file.
+        // In case they are empty we don't add to the command line and we trust in the config file.
+        // In case they are finally missing the fiori deploy toolset will tell us.
+        //
+
+        if (transportRequestId) {
+            params += ['-t', transportRequestId]
+        }
+
+        if (endpoint) {
+            params += ['-u', endpoint]
+        }
+
+        if (abapPackage) {
+            params += ['-p', abapPackage]
+        }
+
+        if (applicationName) {
+            params += ['-n' , applicationName]
+        }
+
+        if (client) {
+            params += ['-l', client]
+        }
+
+        if (desc) {
+            params += ['-e', "\"" + desc + "\""]
+        }
+
+        params += ['-f'] // failfast --> provide return code != 0 in case of any failure
+
+        params += ['-y'] // autoconfirm --> no need to press 'y' key in order to confirm the params and trigger the deployment
+
+        params += ['--username', 'ABAP_USER', '--password', 'ABAP_PASSWORD']
+
+        def fioriDeployCmd = "fiori deploy ${params.join(' ')}"
+        script.echo "Executing deploy command: '${fioriDeployCmd}'"
+        cmd << fioriDeployCmd
+
+        script.withCredentials([script.usernamePassword(
+            credentialsId: credentialsId,
+            passwordVariable: 'password',
+            usernameVariable: 'username')]) {
+
+            /*
+                The config file is configured to read the credentials from the environment,
+                see above in the config file template.
+                After installing the deploy toolset we switch the user. Since we do not su with option '-l' the
+                environment variables are preserved.
+                Not clear of the credentials will also be available as command line parameters.
```

Comment:
[TYPO] Not clear **if**

Could you explain this comment? Could we test it and have an answer?